packages/kanel-enum-tables/src/enumTablesPreRenderHook.ts

```diff
@@ -0,0 +1,191 @@
+import type { TableDetails } from "extract-pg-schema";
+import type {
+  EnumDeclaration,
+  InterfaceDeclaration,
+  Output,
+  PreRenderHook,
+  TypeDeclaration,
+} from "kanel";
+import { resolveType } from "kanel";
+import knex from "knex";
+
+const enumTablesPreRenderHook: PreRenderHook = async (
+  outputAccumulator,
+  instantiatedConfig,
+) => {
+  const connection = instantiatedConfig.connection;
+  const db = knex({ client: "postgres", connection });
+
+  // Get all the tables in all the schemas and filter the ones that have an "$enum" comment
+  const enumTables: TableDetails[] = [];
+  const enumRegex = /(^|\b)@enum(\b|$)/;
+
+  for (const schema of Object.values(instantiatedConfig.schemas)) {
+    for (const table of schema.tables) {
+      if (table.comment !== null && enumRegex.test(table.comment)) {
+        enumTables.push(table);
+      }
+    }
+  }
+
+  const overrides: Output = {};
+
+  for (const table of enumTables) {
+    const primaryKeyColumn = table.columns.find(
+      (column) => column.isPrimaryKey,
+    );
+
+    if (!primaryKeyColumn) {
+      throw new Error(
+        `Table ${table.schemaName}.${table.name} is an enum table but has no primary key`,
+      );
+    }
+
+    const primaryKeyTypeDeclaration = instantiatedConfig.generateIdentifierType(
+      primaryKeyColumn,
+      table,
+      instantiatedConfig,
+    );
+
+    // Get the resolved type for the primary key column
+    const primaryKeyInnerType = resolveType(primaryKeyColumn, table, {
+      ...instantiatedConfig,
+      // Explicitly disable identifier resolution so we get the actual inner type here
+      generateIdentifierType: undefined,
+    });
+
+    if (primaryKeyInnerType !== "string") {
+      // XXX: Should we just warn and skip?
+      throw new Error(
+        `The primary key of enum table ${table.schemaName}.${table.name} must be string-based, got: ${primaryKeyInnerType}`,
+      );
+    }
+
+    const selectorMetadata = instantiatedConfig.getMetadata(
+      table,
+      "selector",
+      instantiatedConfig,
+    );
+
+    const initializerMetadata = instantiatedConfig.getMetadata(
+      table,
+      "initializer",
+      instantiatedConfig,
+    );
+
+    const mutatorMetadata = instantiatedConfig.getMetadata(
+      table,
+      "mutator",
+      instantiatedConfig,
+    );
+
+    // Extract @enumName from the table comment
+    const enumNameMatch = table.comment.match(/@enumName\s+(?<enumName>\S+)/);
```

Comment:
You could use [tagged-comment-parser](https://www.npmjs.com/package/tagged-comment-parser) like I do for the type tags. Not a big deal, but it would give you a bit more flexibility if you want to expand the syntax in the future.

Reply:
Makes sense, will switch.